src/swap/components/SwapSettings.tsx

```diff
@@ -0,0 +1,106 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { cn } from '../../styles/theme';
+import { useBreakpoints } from '../../useBreakpoints';
+import { useIcon } from '../../wallet/hooks/useIcon';
+import type { SwapSettingsReact } from '../types';
+import { SwapSettingsSlippageLayout } from './SwapSettingsSlippageLayout';
+import { SwapSettingsSlippageLayoutBottomSheet } from './SwapSettingsSlippageLayoutBottomSheet';
+
+export function SwapSettings({
+  children,
+  className,
+  icon = 'swapSettings',
+  text = '',
+}: SwapSettingsReact) {
+  const breakpoint = useBreakpoints();
+  const [isOpen, setIsOpen] = useState(false);
+  const [customSlippageEnabled, setCustomSlippageEnabled] = useState(false);
+  const dropdownRef = useRef<HTMLDivElement>(null);
+
+  const handleToggle = useCallback(() => {
+    setIsOpen(!isOpen);
+  }, [isOpen]);
+
+  const handleClickOutsideComponent = useCallback((event: MouseEvent) => {
+    if (
+      dropdownRef.current &&
+      !dropdownRef.current.contains(event.target as Node)
+    ) {
+      setIsOpen(false);
+    }
+  }, []);
+
+  useEffect(() => {
+    document.addEventListener('mousedown', handleClickOutsideComponent);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutsideComponent);
+    };
+  }, [handleClickOutsideComponent]);
+
+  const iconSvg = useIcon({ icon });
+
+  if (!breakpoint) {
```

Comment:
Can you add a line of comment here, I don't understand what are you trying to avoid.

Reply:
yes! This check is following the design in the WalletDropdownBottomSheet. 

I'll test if this is needed, if not then I will remove.